src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,139 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node head;
+    private Node tail;
+    private int size;
+
     @Override
     public void add(T value) {
+        final Node<T> lastNode = tail;
+        final Node<T> newNode = new Node<>(lastNode, value, null);
+        tail = newNode;
+        if (lastNode == null) {
+            head = newNode;
+        } else {
+            lastNode.next = newNode;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index == size) {
+            add(value);
+            return;
+        }
+        checkindexValidity(index);
+        Node<T> newNext = getElement(index);
+        Node<T> newPref = newNext.prev;
+        Node<T> newNode = new Node<>(newPref, value, newNext);
+        newNext.prev = newNode;
+        if (newPref == null) {
+            head = newNode;
+        } else {
+            newPref.next = newNode;
+        }
+        size++;
+    }
+
+    private Node<T> getElement(int index) {
+        Node<T> current;
+        if (index < size / 2) {
+            current = head;
+            for (int i = 0; i < index; i++) {
+                current = current.next;
+            }
+        } else {
+            current = tail;
+            for (int i = size - 1; i > index; i--) {
+                current = current.prev;
+            }
+        }
+        return current;
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T obj : list) {
+            add(obj);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkindexValidity(index);
+        return getElement(index).value;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkindexValidity(index);
+        Node<T> node = getElement(index);
+        T oldValue = node.value;
+        node.value = value;
+        return oldValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkindexValidity(index);
+        return unlink(getElement(index)).value;
     }
 
     @Override
     public boolean remove(T object) {
+        for (Node<T> node = head; node != null; node = node.next) {
+            if (node.value == object || (object != null
+                    && object.equals(node.value))) {
+                unlink(node);
+                return true;
+            }
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private void checkindexValidity(int index) {
+        if (index < 0 || index >= size) {
+            throw new IndexOutOfBoundsException("The index " + index
+                    + " is not valid for size: " + size);
```

Comment:
It`ll be better to make this String constant. I think String.format() can help you!